base/compiler/abstractinterpretation.jl

```diff
@@ -2038,13 +2038,13 @@ function abstract_call_known(interp::AbstractInterpreter, @nospecialize(f),
     elseif la == 2 &&
            (a2 = argtypes[2]; isa(a2, Const)) && (svecval = a2.val; isa(svecval, SimpleVector)) &&
            istopfunction(f, :length)
-        # mark length(::SimpleVector) as @pure
+        # mark length(::SimpleVector) as total
         return CallMeta(Const(length(svecval)), EFFECTS_TOTAL, MethodResultPure())
```

Comment:
Is `MethodResultPure` not still `@pure`? We also still have `is_method_pure/pure_eval_eligible`. Whereas you now want this to be `concrete_eval_eligible`. In any case changing the comment here is misleading.